packages/Search/src/details/DTK_DetailsAlgorithms.hpp

```diff
@@ -45,6 +45,30 @@ bool equals( Sphere const &l, Sphere const &r )
     return equals( l.centroid(), r.centroid() ) && l.radius() == r.radius();
 }
 
+KOKKOS_INLINE_FUNCTION
+bool isValid( Point const &p )
+{
+    using KokkosHelpers::isFinite;
+    for ( int d = 0; d < 3; ++d )
+        if ( !isFinite( p[d] ) )
+            return false;
+    return true;
+}
+
+KOKKOS_INLINE_FUNCTION
+bool isValid( Box const &b )
+{
+    return isValid( b.minCorner() ) && isValid( b.maxCorner() );
```

Comment:
Shouldn't we also check that `minCorner` is less or equal than `maxCorner`?

Reply:
That would invalidate "empty" box returned by default constructor.